ecosystem/typescript/sdk_v2/src/types/hex.ts

```diff
@@ -0,0 +1,91 @@
+import { bytesToHex, hexToBytes } from "@noble/hashes/utils";
+import { HexInput } from ".";
+
+/**
+ * NOTE: Do not use this class when working with account addresses, use AccountAddress.
+ *
+ * NOTE: When accepting hex data as input to a function, prefer to accept HexInput and
+ * then use the static helper methods of this class to convert it into the desired
+ * format. This enables the greatest flexibility for the developer.
+ *
+ * Hex is a helper class for working with hex data. Hex data, when represented as a
+ * string, generally looks like this, for example: 0xaabbcc, 45cd32, etc.
+ *
+ * You might use this class like this:
+ *
+ * ```ts
+ * getTransactionByHash(txnHash: HexInput): Promise<Transaction> {
+ *   const txnHashString = Hex.hexInputToString({ hexInput: txnHash });
+ *   return await getTransactionByHashInner(txnHashString);
+ * }
+ * ```
+ *
+ * This call to `Hex.hexInputToString` converts the HexInput to a hex string with a
+ * leading 0x prefix, regardless of what the input format was.
+ */
+export class Hex {
+  private data: Uint8Array;
+
+  constructor(args: { data: Uint8Array }) {
+    this.data = args.data;
+  }
+
+  static fromString(args: { str: string }): Hex {
+    let input = args.str;
+
+    if (input.startsWith("0x")) {
+      input = input.slice(2);
+    }
+
+    if (input.length === 0) {
+      throw new Error("Hex string is too short, must be at least 1 char long, excluding the optional leading 0x.");
+    }
+
+    return new Hex({ data: hexToBytes(input) });
+  }
+
+  static fromHexInput(args: { hexInput: HexInput }): Hex {
+    if (args.hexInput instanceof Uint8Array) return new Hex({ data: args.hexInput });
+    return Hex.fromString({ str: args.hexInput });
+  }
+
+  toUint8Array(): Uint8Array {
+    return this.data;
+  }
+
+  toStringWithoutPrefix(): string {
+    return bytesToHex(this.data);
+  }
+
+  toString(): string {
+    return `0x${this.toStringWithoutPrefix()}`;
+  }
+
+  static hexInputToBytes(args: { hexInput: HexInput }): Uint8Array {
+    const hex = Hex.fromHexInput(args);
+    return hex.toUint8Array();
+  }
+
+  static hexInputToString(args: { hexInput: HexInput }): string {
+    const hex = Hex.fromHexInput(args);
+    return hex.toString();
+  }
+
+  static hexInputToStringWithoutPrefix(args: { hexInput: HexInput }): string {
+    const hex = Hex.fromHexInput(args);
+    return hex.toStringWithoutPrefix();
+  }
+
+  static isValid(args: { str: string }): { valid: boolean; invalidReason: string | null } {
+    try {
+      Hex.fromString(args);
+      return { valid: true, invalidReason: null };
+    } catch (e) {
+      const error = e as Error;
+      return {
```

Comment:
is there a reason we hold `invalidReasonCode` in AptosAccount but not here? I am not saying we shouldn't have it but wonder why there and not here.

Also, these Objects seem pretty similar here and in AptosAccount, should we create a type of it and define the return type?
```
type Error = {
valid: boolean, invalidReason?: string, invalidReasonCode?: string }
}

static isValid(args: { str: string }): Error
```

Reply:
Well there are only two places where parsing can fail, the check for length and the call to the noble function. So it seemed overkill to have a code. I agree it's more uniform though, let me add a code.